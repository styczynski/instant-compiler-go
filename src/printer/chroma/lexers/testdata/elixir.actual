defmodule Test do
  defmodule Formatter do
    defstruct [:name]
  end

  def hello(name \\ "chroma") do
    case name do
      "chroma" -> IO.puts("Hello, CHROMA!!")
      other -> raise "no others please"
    end
  end
end
