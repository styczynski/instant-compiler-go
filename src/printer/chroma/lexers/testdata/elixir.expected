[
  {"type":"KeywordDeclaration","value":"defmodule"},
  {"type":"Text","value":" "},
  {"type":"NameClass","value":"Test"},
  {"type":"Text","value":" "},
  {"type":"Keyword","value":"do"},
  {"type":"Text","value":"\n  "},
  {"type":"KeywordDeclaration","value":"defmodule"},
  {"type":"Text","value":" "},
  {"type":"NameClass","value":"Formatter"},
  {"type":"Text","value":" "},
  {"type":"Keyword","value":"do"},
  {"type":"Text","value":"\n    "},
  {"type":"KeywordDeclaration","value":"defstruct"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"["},
  {"type":"LiteralStringSymbol","value":":name"},
  {"type":"Punctuation","value":"]"},
  {"type":"Text","value":"\n  "},
  {"type":"Keyword","value":"end"},
  {"type":"Text","value":"\n\n  "},
  {"type":"KeywordDeclaration","value":"def"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"hello"},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"name"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"\\\\"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"chroma\""},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":" "},
  {"type":"Keyword","value":"do"},
  {"type":"Text","value":"\n    "},
  {"type":"Keyword","value":"case"},
  {"type":"Text","value":" "},
  {"type":"Name","value":"name"},
  {"type":"Text","value":" "},
  {"type":"Keyword","value":"do"},
  {"type":"Text","value":"\n      "},
  {"type":"LiteralStringDouble","value":"\"chroma\""},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"-\u003e"},
  {"type":"Text","value":" "},
  {"type":"NameClass","value":"IO"},
  {"type":"Operator","value":"."},
  {"type":"Name","value":"puts"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralStringDouble","value":"\"Hello, CHROMA!!\""},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":"\n      "},
  {"type":"Name","value":"other"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"-\u003e"},
  {"type":"Text","value":" "},
  {"type":"Keyword","value":"raise"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"no others please\""},
  {"type":"Text","value":"\n    "},
  {"type":"Keyword","value":"end"},
  {"type":"Text","value":"\n  "},
  {"type":"Keyword","value":"end"},
  {"type":"Text","value":"\n"},
  {"type":"Keyword","value":"end"},
  {"type":"Text","value":"\n"}
]
